### ISSUE TITLE
Please remove Hexstream's access to Quicklisp and all repositories

### ISSUE THREAD

**Issue by phoe** [quicklisp/quicklisp-projects#1941]
Hello,

I would like to bring your attention to the behavior of @Hexstream that, I assume, might be already well-known to you. 

I regret to inform you that his most recent [GitHub conversation](https://github.com/cl-library-docs/common-lisp-libraries/issues/3), in which I have once more attempted to convince him to change his behavior and publish actual data that shows his real amount and quality of involvement in the Common Lisp community, [I have been announced the main threat to the *whole* Common Lisp community](https://twitter.com/HexstreamSoft/status/1331080126860234752), [declared impossible to directly and indirectly fund in an ethical manner](https://twitter.com/HexstreamSoft/status/1331102289386409985), and then [proclaimed that require immediate medical attention of psychiatric nature](https://twitter.com/HexstreamSoft/status/1331127527977463809).

Hexstream has been ranting on [Twitter](https://twitter.com/HexstreamSoft/) and derailing many [[1]](https://github.com/quickdocs/quickdocs/issues/46) various[[2]](https://github.com/vindarel/lisp-maintainers/pull/1) GitHub[[3]](https://github.com/Hexstream/fakenil/issues/2) discussions[[4]](https://github.com/Hexstream/common-lispers.hexstreamsoft.com/issues/14) at least since 2015. In this time, he has not been stopped by anyone or anything, and this lack of action caused concrete and direct harm to the Common Lisp community.

* He has repeatedly defamed various honored and respected members of the Common Lisp community, including [Rainer Joswig](https://twitter.com/hexstreamsoft/status/1118934532273704960), [Michael Fiano](https://twitter.com/HexstreamSoft/status/1267093005271433218), [Daniel Kochmański](https://twitter.com/HexstreamSoft/status/1171975874276679680), [Stas Boukarev](https://twitter.com/HexstreamSoft/status/1043671920649142272), [Zach Beane](https://twitter.com/HexstreamSoft/status/1054828493467541510), and me. This list is not exhaustive and includes various other Common Lisp programmers on GitHub who have not approved of his behavior.
* He has published a public service announcement detailing his delusion about Zach Beane being a sociopath and a member of the Common Lisp Mafia, and made it available under a custom-purchased domain, https://xach.exposed.
* He fraudulently claims that he is a Common Lisp expert, while [a simple glance at his "extensive contributions"](https://gist.github.com/phoe/ccba343687dc21e3a71d0dc6db68c96e) reveals that they are void, baseless, lack any concrete merit, and completely do not suit the image that he tries to create around himself, which is one of a Common Lisp expert with five years of commercial experience and fifteen years of general Lisp experience.
* He published an online list of Lisp programmers from which he refuses to remove people, because of his made-up policies that he invents and modifies on the spot as appropriate and refuses to consult with anyone.

He is extremely antisocial and abrasive, which further stresses the part of the Common Lisp community that attempts to have fruitful and civil discussion on GitHub.

* He always approaches other people with his arrogant, egocentric point of view, in which his opinion is law, and points of view of others are fully and completely ignorable, and also valid targets for unwinding his aggression.
* Whenever the above arrogance is not respected in any way, he becomes vulgar, aggressive, offensive, obscene, ironic, and derogatory towards his conversation partners.
* When this aggression is pointed out to him and he is pushed to make an answer, he plays the victim card and considers himself to be silenced, censored, and tortured by a so-called "Common Lisp Mafia" that is an invention of his own deluded mind in order to avoid taking responsibility for his actions.
* This behavior of his has been pointed out to him hundreds of times throughout these years, and has not changed, which brings me to the unfortunate but required conclusion, that he is impossible to reason with and must instead be removed from the Common Lisp community by force for the good of our community.
* This kind of behavior directly threatens the healthy future of the Common Lisp community by driving away newcomers to the language, who seek a friendly, inclusive, and non-aggressive environment in which they may learn the language and the programmers.

I can provide more evidence and logs of this behavior on request.

I have written a [blogpost](https://nl.movim.eu/?blog/phoe%40movim.eu/3feaeb8e-0af5-4ddf-b284-5de71ff03804) about this most recent meltdown of his that contains more details and further links.

This kind of behavior has been going on **for years** with multiple members of the Lisp community being targeted. Because of this, it is no longer tolerable in any way. This kind of continuous behavior has been repeated without any hints of change from Hexstream's side require immediate and solid attention in order to protect the Common Lisp community from his influence.

**Therefore, I request immediate revocation of Hexstream's access to the Quicklisp group on GitHub ~~and removal of all of his systems in the next Quicklisp dist.~~** ~~Hexstream's ASDF systems have no non-Hexstream dependents in the current dist, which makes them trivial to remove as a whole.~~ (**Edit:** crossed-out the removal of systems.)

Such action is highly desired and required for the good of the Common Lisp community, as shown by extensive commentary of my discussion with Hexstream on [Reddit](https://www.reddit.com/r/lisp/comments/jzoimg/please_take_a_second_to_show_support_for_uflaming/), [Freenode IRC](https://irclog.tymoon.eu/freenode/%23lisp?around=1606191018#1606191018), [Lisp Discord](https://discord.com/channels/297478281278652417/297478485000192010/780625634607104011), and [Fediverse](https://functional.cafe/@phoe/105263444756143076).

Those discussions in these places mean that the issue with Hexstream is widely known, my means of dealing with him in the main GitHub issue have been appropriate, and the problem has been noticed and acknowledged by tens of members of the Common Lisp community. These people, I assume, would also enjoy a Common Lisp community that is not based upon the pseudo-"ethics" of verbal violence, vulgarity, pretentious meritlessness, paranoidal conspiracy theories, and absolute lack of respect towards other people.

------------------

A matching removal request has been [submitted to the Ultralisp repository](https://github.com/ultralisp/ultralisp/issues/90) given the high importance of this issue.

**Comment by vindarel** [comment]
> removal of all of his systems in the next Quicklisp dist.

I don't see the logical step that leads to this. I find this request weird, violent, illegitimate, inappropriate and counter-productive (he can now claim to be censored).

**Comment by phoe** [comment]
> I don't see the logical step that leads to this.

This is to make a very strong point that indulging aggressive and defamatory behavior towards members of the Common Lisp community for five years straight, as listed above, is not acceptable in any way and has direct consequences. Hexstream's merit is non-zero when it comes to source code, but it is also massively shadowed by the obnoxious behavior that he is infamous for in the Common Lisp community.

> (he can now claim to be censored)

The only censorship that is currently happening is the protection of the Common Lisp community against Hexstream's influence and breaking the continuous streak of five years of his constant obnoxious behavior, as listed above.

If he considers this to be censorship, then so be it; I have run out of any more chances to give him and have decided to instead act to protect the Lisp community from continuous aggression and deluded paranoia of Hexstream's making, since he has proven completely unable to contain it on his own.

**Comment by mfiano** [comment]
While I haven't made contact with him in a couple years, I see that he is still targeting myself and others when things don't go his way. I support this request, not because he bothers me, but because it bothers me that the Lisp community is suffering from his volatile behavior.

**Comment by mohe2015** [comment]
Without being involved in this community recently at all I would also vote for removing his access as he violates all Code of Conducts that exist on the planet and I think it's important that an online community follows some rules and is respectful to each other.

**Comment by thephoeron** [comment]
It is such a gross disappointment to find a thread like this appear in my GitHub notification feed. The behavior I'm witnessing serves no one's interests, whether they are directed back at themselves, or for others.

I'm not going to single anyone out, or take sides, but I'm also not going to dignify persecution with a point-by-point rebuttal. I can't believe I even have to say this much: _it is unethical to request community-wide sanctions be imposed against a peer, for any reason._

It was not long ago I reminded us all, that we Lisp Hackers are meant to have each other's backs. We are also meant to be examples to the greater Lisp community. We don't take the title of Lisp Hacker for ourselves, it is an honorific awarded to those who have given something of significant value to the Lisp community, freely, no matter the cost to themselves. After all, as Lisp Hackers, we are expected to live by the Hacker Ethos: "You're only as good as what you give."

Let us all now ask ourselves, "What have we given to our community, and how much has this dispute taken away from it?"

**Comment by anlsh** [comment]
> It is unethical to request community-wide sanctions be imposed against a peer, for any reason

I have yet to make up my own mind on this matter, and maybe I'll give my own thoughts later if I think of anything worth saying. I can't help but react with incredulity to @thephoeron's statement above though. *Any* reason? Do you mean what you say here? Indeed, do you think community-wide sanctions are ever justified?

**Comment by thephoeron** [comment]
@anlsh: Yes, I mean precisely what I say.

For comparison:
- Quicklisp is maintained by @xach, who is astute, reasonable, fair, forgiving, and informed; he's not a pushover, or off in an ivory tower. So you can count on him to know for himself when someone is violating the code of conduct, and ensure that it is handled correctly.
- @phoe can block anyone he wants from his own GitHub profile. He doesn't even need a reason.
- A comprehensive removal, block, ban, and denial of access from the major quicklisp distributions is far too extreme for one person to request against a peer. That's what makes it unethical, it's a threat of excessive force by someone who has no authority to wield any in the given context.

**Comment by xbrq** [comment]
![screenshot-2020-11-24_18:04:15](https://user-images.githubusercontent.com/69594357/100161570-b5c14a80-2e7f-11eb-8b60-b5a38df171ec.png)

**Comment by lonjil** [comment]
> A comprehensive removal, block, ban, and denial of access from the major quicklisp distributions is far too extreme for one person to request against a peer.

Well, that one person doesn't have any power over it. It is after all just a *request*, and I don't see anything extreme about a person requesting something, since the final decision rests in the hands of everyone else. Xach will see this thread, and come to his own conclusion based on the presented evidence and his own standards.

**Comment by parjanya** [comment]
Never have I seen anything good come from Hexstream dealings in the CL community. On the contrary, the amount of hate and vitriol generated far exceeds any benefit it might have. Banning him might seem a little radical, but I fail to see how would he be missed. What a talent to spread strife, good heavens!

**Comment by thephoeron** [comment]
@lonjil: it's not "just a request", is it? I was being polite and diplomatic above, because we won't help this situation by calling it out for what it is, only fan the flames.

@parjanya: Not that it's anyone's business, but Hexstream is Aspergers, and his reactions to bullying are both normal for Aspergers and perfectly rational.

**Comment by phoe** [comment]
> Let us all now ask ourselves, "What have we given to our community, and how much has this dispute taken away from it?"

I think that the latter part of my [blogpost](https://nl.movim.eu/?blog/phoe%40movim.eu/3feaeb8e-0af5-4ddf-b284-5de71ff03804) might be an answer to the first part of that question; as for the latter, I think it has taken away the possibly false idea that Hexstream's behavior is tolerable or doesn't affect members of the Lisp community that end up meeting him. But, yes, there are obvious other costs associated with discussions like that: time costs spent thinking and writing replies, emotional costs of reading the original posts and rants and then responding on GitHub, et al.

> A comprehensive removal, block, ban, and denial of access from the major quicklisp distributions is far too extreme for one person to request against a peer. 

I do not think that the point that my whole post is trying to make is going to change whatsoever when I rewrite "Hexstream" to "all people who currently have years' worth of notorious defaming, thread-derailing, arrogance, and conspiracy theories involving other Lisp hackers". This is because I can see only one person behaving this way at this moment.

I can agree that in case of only a single case of such extreme behavior it might be troublesome to tell apart witch-hunting and boundary-setting; this is why in the original post of this thread I have first and foremost attempted to list behaviors unacceptable to me and push my issue based on Hexstream being in violation of these behaviors.

> That's what makes it unethical, it's a threat of excessive force by someone who has no authority to wield any in the given context.

Yes, I have no authority here. That is why I have submitted a series of posts to voice my concerns and ask for feedback; finally, I've submit this proposal that I think would solve the issue.

For instance, I have received many comments that my proposal is too wide since it also requires the removal of code; after a night's sleep I consider code to be the most controversial and at the same time the least important and changing part of my proposal. So, sure, let it stay; I have edited the original post.

> Not that it's anyone's business, but Hexstream is Aspergers, and his reactions to bullying are both normal for Aspergers and perfectly rational.

I understand that he has Asperger's and feel deeply sorry for that reason. Still, for the example that started it all I would never call Hexstream's years of ranting against Xach rational, nor would I call Xach's reaction to him bullying in any way. Something that started off as delays in publishing Hexstream's stuff from Xach's side has escalated into something that I cannot rationalize despite my best intentions and lots of effort I've put into it.

For instance, please show me where and how Xach actually bullied him. I can list a single [blogpost](https://lispblog.xach.com/post/127594294213/jean-philippe-paradis-really-really-really-wants) from him and a single [tweet](https://twitter.com/xach/status/635974593400938496) *after* Hexstream has proclaimed to be Xach of the Common Lisp Mafia; afterwards, it was complete radio silence that Xach likely employed in order to protect his mental health, possibly along with social media ignores to not even be aware of Hexstream's rants. To count in additional fallout afterwards, the now-mafioso Xach was seemingly capable of influencing Robert Strandh's opinions about Hexstream's MOP rendition[[1]](https://twitter.com/HexstreamSoft/status/979366847618453504) [[2]](https://twitter.com/HexstreamSoft/status/1019779534223224832); I won't dive further into Twitter tweets. Seriously, how is the original reaction from Xach bullying, how is this whole reaction from Hexstream rational at all, and how is this whole reaction from Hexstream not a massive, years-long, aggressive attack in order to smear feces all over Xach's name due to Hexstream's delusions?

I say delusions because Hexstream seems to fill in the blanks in his thoughts with a lot of false assumptions about other people's intentions and proceed with his line of thinking without ever paying any attention or care to verifying them. Seriously, I have never seen him seriously ask for rationale about why someone behaved this way or that way. For instance, he knows better than me why I have written all of this content about him; I am mentally ill and require immediate psychiatric help.

To me, this is delusion in its purest form, which injects pure irrationality into the otherwise strictly rational process that I was able to observe in Hexstream from outside. If I understand Asperger's correctly, then reading other people's intentions is seriously impaired in this way; but, instead of generating conspiracy theories based on fear, one can instead *ask* why other people behave in some way, and seriously, they'll almost always respond in honesty, without ill intentions, and also with an apology if they notice that they have screwed up somewhere along the way.

It is going to be different now with Hexstream since other people can, and will, assume ill intentions on his side based on all the behavior I have listed in the original issue and the blogpost, but still, with a little bit of work and an apology that he was blind to the fact that he was blind, even he, in his current moment, can get a constant, reliable source of information about other people's intentions that will work much, much better than what seems to be a pseudorandom malicious intention generator.

Seriously, despite all that I've seen and read from him and despite already written on the topic, I feel myself unable to demonize Hexstream altogether and still consider him to be capable of change. IMO, the first and most important change that would also give the greatest benefit would be to feed his otherwise working and rational mind with better sources of information. Sure, this requires trust in other people that he does not seem to have to overcome fear that is one of the emotions that bind the strongest, but he is not required to work on this purely on his own and can instead reach for support. (Saying this as a veteran of seven years of psychotherapy.)

Frankly, at this moment, other people do not seem to trust Hexstream much either, but this will change when and if they see a basis for change. I don't think members of the Lisp community who've voiced their concerns here or in Reddit fail to understand that minds are complex beasts, can and will play tricks and work in varying ways, and that they can be very tough to tame in general; I think they are tired of the rants and abrasions that Hexstream is currently infamous for.

> It was not long ago I reminded us all, that we Lisp Hackers are meant to have each other's backs. We are also meant to be examples to the greater Lisp community.

As I have mentioned in one of the earlier posts somewhere in the threads that have popped up, I cannot have the back of anyone who repeatedly attempts to smack the backs of other people.

But, you know, I'm still having Hexstream's back. This block of text above is mostly how I understand his current problems and what I think, with my best intentions, would be the way forwards and out of them.

And, at the same time, I'm still having the backs of other Lisp hackers, which is why I still strongly vote for isolating Hexstream in his current state of behavior to avoid further harm from becoming possible.

**Comment by mohe2015** [comment]
Considering that it came up now https://mobile.twitter.com/HexstreamSoft/status/1154310860199931906 is the original source which also mentions OCPD aside from Asperger's. It may be an interesting read for everyone but I don't want to say anything more to this as I don't know the severeness and I think he can talk for himself if he wants to. I can also remove this message if you don't want this to get more publicity.

I'm quite an outsider but to me it's unclear how this all started. I read a few threads from GitHub and most of them don't seem to match with other threads and the tweets (I also realized I had a short conversation with him in the past). It seems like there is some intermediate piece missing.

**Comment by tfeb** [comment]
Please don't do this, or at least not because of this issue (or this comment).  The CL community has a very long history of difficult people, including me.  The solution to those people is to not engage with them: just walk away.  If some difficult person thinks they are being persecuted by some mythical Common Lisp Mafia then when a bunch of Common Lisp people *get together and persecute them* this is not going to help.

Instead, well, all these new-fangled things have killfiles: you can block users on GitHub,  you can I am sure block people on Twitter, you can ignore people on IRC.  Just walk away.

(Incidentally, I can state with some certainty that the Common Lisp Mafia has not, in fact, persecuted this person, as I have not authorised such action.  Our recent activities have been directed elsewhere.)

**Comment by thephoeron** [comment]
@mohe2015: it's not about avoiding publicity, or keeping secrets, or even the right to privacy in this case because as you pointed out, Hexstream has publicly shared that he is on the spectrum, with the complication of OCPD. My concern is the importance of avoiding discrimination, whether intentional or unintentional.

The fact is, there is no intrinsic problem or deficiency in individuals on the Autism spectrum; the behavioral problems and severity associated with the Autism spectrum are the result of abuse, neglect, isolation, bullying, and a life of constant communication blocks and misunderstandings, to say nothing of the nearly complete failure to provide for their needs during critical phases of early childhood development. The Autism spectrum is nothing more than evolution presenting our species with a new, and fundamentally different neurological model. Unfortunately, humans are primates, and primates are xenophobic. Human psychology even has built-in safeguards to protect and reinforce xenophobia. Luckily, human psychology is adaptable, so with patience, humility, discipline, and reinforcement, we can learn to listen to our better angels instead of the beast within.

As an aside of general interest, the principal construct responsible for protecting xenophobia is the Ego.

@phoe: your treatment of Hexstream is vicious and cruel, however, I am aware that it is a projection of your own feelings of self-worth, so you have my deepest sympathy for the struggle you have to live with. These extensive posts you've been writing, their length, tone, meter, as well as the structure of your reasoning process appear to be early manifestations of an impending psychotic break, however, the persecutory fixation and demonization of one of your closest friends in the community are mostly still sub-delusional and not paranoid. I hope it is not too late for you to hear me when I say: you need professional help, and there's no shame in admitting it or seeking it. Please see a psychiatrist immediately, and let them help you take care of yourself. The mind is all you've got.

**Comment by phoe** [comment]
@thephoeron Deepest thanks to you for your worry; I am already under weekly care of a psychotherapist and less frequent, but consistent, care of a psychiatrist.

I agree with your definition of where the behavioral problems come from; the same is true in general, not just for people on the Autism spectrum; this must also be the case for Hexstream. That's why, as a compassionate human being, in total, I posted tens of sentences that provide data for him in hope that he's capable of using it to perform some self-introspection, analyze his current behavior and the way its various parts stem from his past, and in effect reduce the suffering he receives from Lisp hackers in general. If you read into the messages I've posted in the earlier thread about arrows, you should be able to see these. They might be perceived by Hexstream as painful, because they require trust into other people; they likely also collide with some parts of Hexstream's worldview and therefore, if applied, would force him to change the shape of his personality that was molded in his earlier years (which is a painful process, as I can testify myself). Still, they are also posted by me in good faith and benevolence, in simple hope that they are useful and can ease his suffering.

As the same compassionate human being that I am, I am also a part of a system that is my chunk of the Common Lisp community, I need to combine the above with caring for the group. To the best of my ability, I cannot willingly allow anyone's actions to negatively affect other members of the group that I care about. This applies in all situations, even if that person is also a member of this group, or when the reasons for those negative actions stem from one's past treatment and are therefore not of their fault. In such a case, I am deeply sorry and compassionate about the reasons that led to such behavioral problems, but I must also be strict and ruthless about the actions themselves and their consequences. If I understand correctly, then this latter part is exactly what you perceive as cruel and vicious; I hope that this post provides you with more context into why I am doing what I am doing.

I deeply respect your support for Hexstream and I consider it worth its weight in gold. I would like to be able to provide such as well, but, as mentioned earlier, I can only provide it conditionally - with the condition being him stopping his acts of aggression and making appropriate amends, regardless of what are the reasons for such kind of behavior. I am not a specialist and do not have the resources help him with working on his way towards this condition, but I think that Hexstream, who based on his GitHub information lives in a sizeable town, has proper access to external resources for achieving this.

Also, to be clear, I do not act out of xenophobia, since it is not evolution itself that I am against; I act against behavior that I consider destructive to me and Lisp hackers I care and worry about in order to not let it produce even more suffering. I hope that this is understandable.

(Sorry about another wall of text; I am trying to ensure that I'm understood clearly.)

------------

> the persecutory fixation and demonization of one of your closest friends in the community are mostly still sub-delusional and not paranoid

I am sorry; I did not understand this part of your sentence. Could you elaborate a bit?

**Comment by phoe** [comment]
@thephoeron I also find it nontrivial to view this current issue from the evolutional point of view, therefore I allowed myself to write a little bit more about it. Hence the second post; separate, because it's a separate issue.

The process of evolution, when viewed from the perspective of a single entity, is generally two-fold: the first step is the creation of an entity with random mutations to it, the second - examining the entity's fitness in whatever environment it functions in. The second part is continuous, and therefore self-introspection and self-modification is possible for self-aware agents; the first one, is, unfortunately, mostly atomic and unchangeable.

A very close topic is evolution of whole systems of entities, which in case of humans are various societies, groups, and communities. (Have you read [Thinking in Systems](https://www.amazon.com/Thinking-Systems-Donella-H-Meadows/dp/1603580557)?) Each such system encounters a fundamental problem of defining what kinds of behavior are wanted and unwanted in it, which naturally establishes a fitness subfunction of the system itself, executed concurrently with the system members' own functions. 

This group function depends on the choices made by the individuals the system is composed of, which hooks directly into the point you make about the adaptability of human psychology. It usually involves some kind of self-preservation of the group, its members, and its resources. (In case of self-aware actors, it also sometimes manifests as an artifact of its own; that's how all kinds of written laws and bylaws and codes of conduct come into existence.)

Still, evolution is a process of choice, and evolution of groups is no different. Individuals within each group make decisions, and these decisions are perceived by other members of it. This constantly reevaluates the base question of which exactly behaviors are acceptable within a group and which are not. It is an important question because in the general case accepting too much change is unwanted, since in such a case a destructive mutations have a chance of destroying a group from the inside; rejecting too much change is also unwanted, since it weakens the group's adaptability to internal and external challenges.

So, in this context, the main question is: as an individual self-aware entity and an element of the system consisting of Common Lisp programmers, what am I supposed to do with Hexstream's claims that one person [is a part of the Common Lisp mafia](https://twitter.com/HexstreamSoft/status/985486505677291520), another person [is a low-functioning sociopath](https://twitter.com/HexstreamSoft/status/1267093005271433218), yet another person's depression [is an "act"](https://twitter.com/HexstreamSoft/status/665241451991814144), yet yet another person [is clinically insane](https://twitter.com/HexstreamSoft/status/1118733322602057734), yet yet yet another person [allegedly panicked and locked a GitHub thread](https://twitter.com/HexstreamSoft/status/1134931101607239680), and so on? Do I maybe accept them and welcome them into the system, either via passive inaction or via explicit approval? On what basis? How do I rationalize that to people who are negatively affected by these actions? How do I argument my decision against people who think otherwise? Do I consider them to be harmful and abrasive to me? How about other entities in the system? How about the system as an entity of its own and how it is perceived in the metasystem being the broader programming environments of the Lisp family? How about the even broader metametasystem of the general programming community, e.g. on GitHub? Do I maybe reject these actions? How do I shape my rejection? How do I exclaim it?

This is the sort of questions that I needed to answer when I encountered Hexstream's derailment of the arrows thread and that I have to reevaluate continuously ever since when I post anything related to this whole issue. I hope that I've clarified. I've written it to show you the way I think and perceive this issue; if you see anything in it that could be improved, I'll gladly listen.

**Comment by thephoeron** [comment]
@phoe: this is a lot of material to review and respond to, so I'll have to get back to most of it later. I'm glad that you have the benefit of weekly psychotherapy, especially with current conditions of the global pandemic steadily worsening.

In psychotherapy, do you work through the exercise of reviewing your decisions together?

One observation I will address now, a significant source of contention may be related to a conflict of value systems, such as your statement: "I cannot willingly allow anyone's actions to negatively affect other members of the group I care about." It seems to imply that you believe you have both the moral authority and de-facto responsibility to impose any force deemed necessary to classify and control the actions of others, when from the Canadian perspective, such a responsibility conflicts with the Hacker Ethos, and in general an imposition of force voids all moral authority. On the occasions when force is ethically necessitated, it is still a failure of diplomacy, which in turn typically stems from miscommunication. But I am fairly certain that judging others under one's own value system while ignoring theirs is always unethical. Most importantly, the only ethical reason to call someone out for being unethical is to offer them a learning opportunity, not to "teach them a lesson".

**Comment by phoe** [comment]
@thephoeron No problem; please take your time. The pandemic has spared me and mostly spared my surroundings, so things are good so far. No, we do not practice such a thing on every single psychotherapy session, but I am free to touch any topic I want to; I will bring this current one up and discuss on the next session.

I'll also take a break and reply tomorrow, or when you post back with a full reply, whichever comes later.

**Comment by thephoeron** [comment]
@phoe: thanks. I may need a few days before I can get back to this, though.

This is an incredibly stressful situation, not only because of the high visibility of everyone involved, nor the precedents set for the entire Lisp community by the way we resolve this issue, nor even the ethical challenges we are uncovering that we may not yet have correct answers for, but still need to consider to remain true to ourselves and each other.

As a Veteran of the Canadian Armed Forces with PTSD, it is actually quite traumatic for me to step up and be a peacekeeper in the community, but I'm doing it anyway.

**Comment by phoe** [comment]
@thephoeron yes, thank you for your work, even more so because it is exhausting for you to perform it. As tough as it is for me to say it, maybe it is because you've already seen some real shit that you are capable of attempting to grasp this situation and digest it, no matter how long as it takes you (and a few days for such an amount of material is, IMO, not long by any standards).

I hope that collectively we figure out a beneficial and clear way forward. It'll take some work, and pain, and learning, and likely taking some steps backwards before we can move forth again. Which is ultimately good, because then we'll avoid repeating some mistakes in the future.

I'll let @quicklisp decide whether and for how long this issue should stay open; I think that everything that needed to be said for the time being was already said, along with some things that did not need to be said but nonetheless were.

Possibly now is the time to get some peace and think about the stuff that happened; therefore, again, please take all the time you need.

**Comment by thephoeron** [comment]
@phoe: Exhaustion is secondary, a side-effect I didn't consider important enough to mention. I do mean _traumatic_. But doing nothing would be worse. So can I ask, for the sake of the greater good, that you rescind this request from Quicklisp and Ultralisp?

**Comment by phoe** [comment]
@thephoeron OK. Closing.

Ultralisp has declined my request, so there is nothing to rescind there.

**Comment by thephoeron** [comment]
@phoe: there are so many interesting topics that have come up in this thread. Valuable explorations of ethics under an inclusive, multi-cultural, generalist paradigm. Evolutionary vs Systems-level analysis. I could talk about "Thinking In Systems" for a week straight, myself!

So that gets me wondering. Is there a better place to explore potentially expansive threads like this than the Quicklisp Projects issues board? It reminds me of Usenet before AOL, and the Bulletin Board Systems when I was a kid. Is there an existing Common Lisp community forum, or is that something I could set up that would be valuable enough to the community that Black Brane should sponsor it?

**Comment by phoe** [comment]
I'd suggest comp.lang.lisp if only it wasn't a [spambot- and troll-filled cesspool](https://groups.google.com/g/comp.lang.lisp) nowadays; it's sad that it ended up going like this. I'm not aware of any other public Lisp newsgroups that would be in a better shape while still being active; maybe someone else knows. [LispForum](https://www.lispforum.com/) is dead and has been inactive for a long time.

I don't know any better forum that would facilitate such discussion. I sorta wish there was. If we limit ourselves to the tools that we have available, maybe we could pretend that GitHub issues are suitable for that, open up a new repository, and discuss there? If we want to invest a little bit of time, maybe we should open up a forum, e.g. a [Discourse](https://www.discourse.org/) instance, for Lispers to talk on? Maybe we should open up a NNTP c.l.l alternative that does not fall prey to spam like the original did, in order to pay homage to all the people who use the news protocol and consume discussion threads mostly via mail?

I honestly don't know. It seems like the good times of public usenet and BBSes are over, so whatever place for public discussion we end up having is going to be of our own making and moderating.

For now, I'd propose the issues section of a new, separate GitHub repository, simply because it's the cheapest option. If you want to invest some resources in it, a Discourse instance (possibly with a [NNTP bridge](https://github.com/sman591/discourse-nntp-bridge)) would IMO be the best choice.

**Comment by cage2** [comment]
On Mon, Nov 30, 2020 at 12:47:07AM -0800, phoe wrote:

Hi everyone!

> I'd suggest comp.lang.lisp if only it wasn't a [spambot- and troll-filled cesspool](https://groups.google.com/g/comp.lang.lisp) nowadays; it's sad that it ended up going like this. I'm not aware of any other public Lisp newsgroups that would be in a better shape while still being active; maybe someone else knows. [LispForum](https://www.lispforum.com/) is dead and has been inactive for a long time.
>
> I don't know any better forum that would facilitate such discussion. I sorta wish there was.

Why   not  a   discourse   (https://www.discourse.org/)  instance   on
common-lisp.net?  Not  optimal but  could works someway  as it  try to
promote responsible use (user levels,  moderation etc) by the users, i
miss a platform for discussing CL not tied to surveillance capitalism,
IRC is fun and pretty good but  too "fast" for people like me with low
English skill and low attention threshold. :)

NNTP is out of discussion for me for well known motivations. :)

Bye!
C.
